yb-voyager/cmd/importData.go

```diff
@@ -848,39 +848,39 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 					}
 				}
 				res, err := conn.PgConn().CopyFrom(context.Background(), reader, copyCommand)
-				rowsCount = res.RowsAffected()
+				rowsAffected = res.RowsAffected()
 
-				if err != nil {
+				if err != nil && strings.Contains(err.Error(), "invalid input syntax") {
+					return false, err
+				}
+
+				if err != nil || (rowsAffected != (task.OffsetEnd - task.OffsetStart)) {
 					log.Warnf("COPY FROM file %q: %s", inProgressFilePath, err)
-					if !strings.Contains(err.Error(), "violates unique constraint") {
+					if err != nil {
 						log.Errorf("RETRYING.. COPY %q FROM file %q due to encountered error: %v ", task.TableName, inProgressFilePath, err)
-						duration := time.Duration(math.Min(MAX_SLEEP_SECOND, math.Pow(2, float64(COPY_MAX_RETRY_COUNT+1-copyRetryCount))))
-						log.Infof("sleep for duration %d before retrying...", duration)
-						time.Sleep(time.Second * duration) // delay for 1 sec before retrying
-						copyRetryCount--
-						return copyRetryCount > 0, err
+					} else {
+						log.Errorf("RETRYING.. COPY %q FROM file %q since rows affected(%d) doesn't match actual row count(%d)",
```

Comment:
If a user retries after deleting some row(s) from a batch, yb-voyager will never be able to mark the batch as completed!!!

Add a comment here in the code AND file a ticket. We will have to fix the issue before GA.

Reply:
Done - #223